### DISCUSSION TITLE
是否有android系统上和mars-xlog的性能对比？

### DISCUSSION THREAD

**Discussion by ailichao** [Tencent/BqLog#5] [General]
是否有android系统上和mars-xlog的性能对比？

如压缩率、耗时等

https://github.com/Tencent/mars/tree/master/mars/xlog

**Comment by pippocao** [comment]
xlog的优势在于加密，内存中没有明文等，性能和实时高性能压缩方面相对不是它的长项。下周可以做一个评测。

**Comment by pippocao** [comment]
测试完了，结论就不贴了，性能有巨大的，数量级上的差距，其中xlog用的是异步日志和ZLIB_MODE。 

原因也比较清晰，有比较多
1. xlog并没有做format的处理，而是用了默认的java的String.format
2. IO层面没有做极限优化
3. 压缩格式是通用压缩算法，而不是写的时候的特殊定制格式，所以压缩带来的不是性能的提升，反而是性能的损耗
4. 并发的总线性能优化不够极限
5. 整体代码不是朝着极端性能去做的
